plugins/foreman_discovery/3.0/index.md

```diff
@@ -0,0 +1,689 @@
+
+To install the plugin, do
+
+    # yum install rubygem-smart_proxy_discovery
+
+on Fedora and Red Hat systems or
+
+    # apt-get install ruby-smart-proxy-discovery
+
+on Debian and Ubuntu systems.
+
+Foreman Proxy must be restarted after plugin installation and features must be
+refreshed for the given Proxy in the Foreman interface (*Infrastructure >
+Smart Proxies > a proxy > Refresh features*).
+
+Foreman Proxy Discovery plugin installation can be skipped for "legacy"
+operation when nodes directly talk to Foreman and vice versa.
+
+## 2.3 Foreman Discovery Image
+
+The image is based on CentOS 7 and about 150 MB in size. There are two ways
+to install the image.
+
+### 2.3.1 Download via installer (recommended)
+
+As of Foreman 1.8+, the foreman-installer is able to automatically download
+latest stable image. For this, re-run the installer with the following option:
+
+    # foreman-installer \
+        --foreman-plugin-discovery-install-images=true
+
+Tip: It is possible to install both Discovery plugin and image in one
+installer run by providing both the options.
+
+Important note: Executing foreman-installer will re-deploy all foreman-related
+configuration files. In case some changes have been made, proceed with manual
+download described below.
+
+### 2.3.2 Manual download
+
+Images are available [for direct
+download](http://downloads.theforeman.org/discovery/)
+To download the latest release to the expected location, do the following on
+Fedora and Red Hat systems:
+
+    # wget http://downloads.theforeman.org/discovery/releases/latest/fdi-image-latest.tar \
+      -O - | tar x --overwrite -C /var/lib/tftpboot/boot
+
+On Debian systems, use `/srv/tftp/boot` instead of `/var/tftpboot/boot`.
+
+### 2.3.3 Verify checksums
+
+    # cat /var/lib/tftpboot/boot/fdi-image/SHA256SUM
+    beb3cfba7d9fb9d71481c0c8f... initrd0.img
+    f03bce150d2473a28697d2988... vmlinuz0
+    # sha256sum /var/lib/tftpboot/boot/fdi-image/*
+
+On Debian systems, use `/srv/tftp/boot` instead of `/var/tftpboot/boot`.
+
+### 2.3.4 Building own image
+
+To build a discovery image, please visit the foreman-discovery-image git
+[repository](https://github.com/theforeman/foreman-discovery-image) and find
+the README for further instructions.
+
+## 2.4 Hammer CLI Foreman Discovery
+
+### 2.4.1 Installation
+
+Before installing make sure you have installed the [hammer_cli_foreman plugin](https://github.com/theforeman/hammer-cli/blob/master/doc/installation.md),
+
+#### 2.4.1.1 RPM
+
+```yum install rubygem-hammer_cli_foreman_discovery```
+
+#### 2.4.1.2 GEM
+
+```gem install hammer_cli_foreman_discovery```.
+
+## 2.5 Upgrade
+
+To upgrade Foreman Discovery follow the standard procedure of upgrading all
+the Foreman packages.
+
+New component Proxy Discovery plugin was introduced for Discovery 2.0
+therefore there are multiple ways how to re-configure existing installation.
+
+### 2.5.1 Upgrade without Proxy Discovery plugin (recommended)
```

Comment:
> no longer recommended?

Yeah, I wanted to wait one additional release with that, but I've
doublechecked and there is no need to wait. Changed everything - we now
default to proxy configuration.

-- 
Later,
 Lukas #lzap Zapletal